pages/table/sticky-columns.page.tsx

```diff
@@ -0,0 +1,318 @@
+                errorIconAriaLabel: 'Edit cell last error',
+                editingCell: (item, { currentValue, setValue }) => {
+                  return (
+                    <Input
+                      autoFocus={true}
+                      value={currentValue ?? item.name}
+                      onChange={event => setValue(event.detail.value)}
+                    />
+                  );
+                },
+              },
+            },
+          ]}
+          items={ITEMS}
+          ariaLabels={{
+            activateEditLabel: column => `Edit ${column.header}`,
+            cancelEditLabel: column => `Cancel editing ${column.header}`,
+            submitEditLabel: column => `Submit edit ${column.header}`,
+            allItemsSelectionLabel: ({ selectedItems }) =>
+              `${selectedItems.length} ${selectedItems.length === 1 ? 'item' : 'items'} selected`,
+            itemSelectionLabel: ({ selectedItems }, item) => {
+              const isItemSelected = selectedItems.filter(i => i.name === item.name).length;
+              return `${item.name} is ${isItemSelected ? '' : 'not'} selected`;
+            },
+            tableLabel: 'Inline editing',
+          }}
+          sortingDisabled={true}
+          header={<Header>Inline editing columns</Header>}
+        />
+        <Table
+          data-test-id="selection-single"
+          selectionType="single"
+          stickyColumns={{ first: 1, last: 1 }}
+          columnDefinitions={COLUMN_DEFINITIONS}
+          items={ITEMS}
+          sortingDisabled={true}
+          ariaLabels={{
+            selectionGroupLabel: 'Items selection',
+            allItemsSelectionLabel: ({ selectedItems }) =>
+              `${selectedItems.length} ${selectedItems.length === 1 ? 'item' : 'items'} selected`,
+            itemSelectionLabel: ({ selectedItems }, item) => {
+              const isItemSelected = selectedItems.filter(i => i.name === item.name).length;
+              return `${item.name} is ${isItemSelected ? '' : 'not'} selected`;
+            },
+            tableLabel: 'Selection type single',
+          }}
+          header={<Header>Selection type single</Header>}
+        />
+        <Table
+          data-test-id="selection-multi"
+          selectionType="multi"
+          stickyColumns={{ first: 1, last: 1 }}
+          columnDefinitions={COLUMN_DEFINITIONS}
+          items={ITEMS}
+          sortingDisabled={true}
+          ariaLabels={{
+            selectionGroupLabel: 'Items selection',
+            allItemsSelectionLabel: ({ selectedItems }) =>
+              `${selectedItems.length} ${selectedItems.length === 1 ? 'item' : 'items'} selected`,
+            itemSelectionLabel: ({ selectedItems }, item) => {
+              const isItemSelected = selectedItems.filter(i => i.name === item.name).length;
+              return `${item.name} is ${isItemSelected ? '' : 'not'} selected`;
+            },
+            tableLabel: 'With 5 sticky columns',
```

Comment:
This example can be removed if making preferences shared between tables.

Reply:
I managed to break the styles by using compact mode for the table with 5 sticky columns. There is a gap between the 2 and 3 columns.

<img width="1173" alt="Screenshot 2023-04-18 at 14 40 02" src="https://user-images.githubusercontent.com/20790937/232780193-d0732604-b075-47fa-86cf-58d0b0ab04a5.png">

Reply:
And also between 4 and 5 columns (I didn't notice that at first).

Reply:
I can reliably reproduce this in Chrome and Firefox, but not in Safari.

Reply:
In Firefox I also occasionally noticed small 1px gaps between 1 and 2 sticky columns and after the last sticky column in comfortable and compact modes.

Reply:
I've added an effect for recalculating offset widths on the wrapper's scroll width change - this should be fixed.

Reply:
Regarding the 1px gap, I think I found the problem:

The offset is a non-integer number of pixels, when computing offsetWidth, the real width is rounded to nearest integer. I probably need to use getBoundingClientRect().width to avoid this silly rounding

Reply:
The gap problem is still present. You can see in dev tools that the cumulative width of the first two sticky columns is about 525px. The left offset of the 3rd columns is 533px.

https://user-images.githubusercontent.com/20790937/233009656-ca39fce4-6d84-425a-a5ce-b0b059e3781c.mov